File: Bronze_JavaScript/2231.js
// 분해합

const fs = require("fs");
const input = fs.readFileSync("/dev/stdin").toString().trim();

const num = parseInt(input);
let result = 0;

for (let i = 0; i <= num; i++) {
  let sum = i;
  const numList = String(i).split("").map(Number);

  sum += numList.reduce((acc, curr) => acc + curr, 0);

  if (sum === num) {
    result = i;
    console.log(result);
    break;
  }
}

// 조건을 만족하는 생성자가 없는 경우
if (result === 0) {
  console.log(0);
}

File: Bronze_python/2231.py
# 분해합

n = int(input())
result = 0

for i in range(1, n+1):
  num = list(map(int, str(i)))  # sum(num)은 리스트 요소를 모두 더할 수 있음
  result = i + sum(num) # i와 i의 각 자리 수를 더함

  if result == n:
    print(i)
    break

  if i == n:
    print(0)